From 67e5c78e41e7fe879bed01d65943fd3205e7ad3c Mon Sep 17 00:00:00 2001
From: "jaekyu.lee" <jaekyu.lee@lge.com>
Date: Thu, 23 Jan 2025 16:30:57 +0900
Subject: [PATCH] Fix build with liburcu-0.15.0

:Release Notes:
Fix build error

:Detailed Notes:
Fixes:
http://gecko.lge.com:8000/Errors/Details/1095642

FAILED: src/core/trace/CMakeFiles/WebAppMgrTrace.dir/web_app_manager_tracer_lib.cc.o
...
In file included from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/wam-clang/1.0.2-99/git/src/core/trace/web_app_manager_tracer_lib.cc:19:
In file included from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/wam-clang/1.0.2-99/git/src/core/trace/pmtrace_webappmanager3_provider.h:31:
In file included from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/wam-clang/1.0.2-99/recipe-sysroot/usr/include/lttng/tracepoint.h:13:
In file included from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/wam-clang/1.0.2-99/recipe-sysroot/usr/include/lttng/tracepoint-rcu.h:10:
TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/wam-clang/1.0.2-99/recipe-sysroot/usr/include/urcu/compiler.h:205:1: error: templates must have C++ linkage

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-17603] Fix build with liburcu-0.15.0

Change-Id: I7467077afb1f7ab27bd75bd70426071c5b82fe32
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/wam/+/466523]

 src/core/trace/pmtrace_webappmanager3_provider.h | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/src/core/trace/pmtrace_webappmanager3_provider.h b/src/core/trace/pmtrace_webappmanager3_provider.h
index 631c30db..af28fcc0 100644
--- a/src/core/trace/pmtrace_webappmanager3_provider.h
+++ b/src/core/trace/pmtrace_webappmanager3_provider.h
@@ -20,16 +20,16 @@
 #undef TRACEPOINT_INCLUDE
 #define TRACEPOINT_INCLUDE "pmtrace_webappmanager3_provider.h"
 
-#ifdef __cplusplus
-extern "C" {
-#endif /*__cplusplus */
-
 #if !defined(_PMTRACE_WEBAPPMANAGER3_PROVIDER_H) || \
     defined(TRACEPOINT_HEADER_MULTI_READ)
 #define _PMTRACE_WEBAPPMANAGER3_PROVIDER_H
 
 #include <lttng/tracepoint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif /*__cplusplus */
+
 /* "message" tracepoint should be used for single event trace points */
 TRACEPOINT_EVENT(pmtrace_webappmanager3,
                  message,
@@ -83,11 +83,10 @@ TRACEPOINT_EVENT(pmtrace_webappmanager3,
                  function_exit,
                  TP_ARGS(const char*, text),
                  TP_FIELDS(ctf_string(scope, text)))
+#ifdef __cplusplus
+}
+#endif /*__cplusplus */
 
 #endif /* _PMTRACE_WEBAPPMANAGER3_PROVIDER_H */
 
 #include <lttng/tracepoint-event.h>
-
-#ifdef __cplusplus
-}
-#endif /*__cplusplus */
